hibernate-core/src/main/java/org/hibernate/jpa/spi/CriteriaQueryTupleTransformer.java

```diff
@@ -112,7 +116,7 @@ public <X> X get(String alias, Class<X> type) {
 					);
 				}
 			}
-			return (X) untyped;
+			return type.cast(untyped);
```

Comment:
The change seems unnecessary, please revert it

Reply:
The cast to `X` generates a warning that there is an unchecked cast, while `type.cast(untyped)` does not. The `NativeTupleImpl#get(String, Class<X>)` method does it like that as well. I can revert it but that brings back the warning.